Move resolve script to actions

bodubas opened the pull request.

bodubas committed:
Change resolve script to actions

bodubas committed:
Add actions image

bodubas committed:
Update Property

bodubas committed:
Update rate limit

bodubas commented on CiscoVM/CiscoVM-1.0.0/ciscovm_helpers.py:
```diff
@@ -53,44 +54,31 @@ def _generate_headers(self, event_type: str = CHECK_EVENT_TYPE) -> dict:
-
-            logging.debug(msg)
-            return False, msg
+                response = func(self, *args, **kwargs)
```
Removed error handling logic to avoid redundan code in other modules.

We percieve error response code the same as any other exception, so if we don't raise it here, it causes duplication in test and export scripts.

bodubas commented on CiscoVM/CiscoVM-1.0.0/ciscovm_reset.py:
```diff
@@ -20,30 +20,16 @@
 SOFTWARE.
```
This action allows to reset hash and status to reexport the device. It's going to be done monthly and on demand.

bodubas commented on CiscoVM/CiscoVM-1.0.0/ciscovm_test.py:
```diff
@@ -33,17 +33,11 @@
-            "succeeded": True,
-            "result_msg": f"Connected to '{client.full_url}'.",
-        }
-    else:
```
Removed "else" clause as client throws exception on bad request.

bodubas commented on CiscoVM/CiscoVM-1.0.0/system.conf:
```diff
@@ -34,17 +34,17 @@
         {
           "rate limiter": true,
           "display": "Rate limiter",
-          "unit": 6,
```
Decreased to 1 second interval limit to have better reqeusts count distribution.

mstoro commented:
I think according to other eyeExtend apps and [this dir generating tool](https://github.com/fs-connect/connect-images-dir/tree/master) we need to add a set of icons (connect, failed, gray, waiting) for all the actions

sveta-trotsiuk commented on CiscoVM/CiscoVM-1.0.0/ciscovm_helpers.py:
```diff
@@ -53,44 +54,31 @@ def _generate_headers(self, event_type: str = CHECK_EVENT_TYPE) -> dict:
+                if response.status_code in self.RETRY_STRATEGY["status_forcelist"]:
+                    # Retry request with backoff delay
+                    delay = self.RETRY_STRATEGY["backoff_factor"] * (2 ** attempt)
+                    msg = f"Status code: {response.status_code}"
+                    logging.debug(f"{msg}. Retrying in {delay} seconds...")
```
Lets remove msg var

bodubas replied:
Actually, should we remove that? Basically, we suppress error here fore retry purpose, so logging is the only way to know if the retry happend and why.

sveta-trotsiuk commented on CiscoVM/CiscoVM-1.0.0/property.conf:
```diff
@@ -9,50 +9,78 @@
+      "label": "Cisco VM Export",
+      "group": "connect_ciscovm_actions",
+      "description": "Export assets to Cisco VM",
+      "ip_required": false,
```
I'm wondering. Why not required?

bodubas replied:
Based on our mapping rules, mac address is enough, so missing ip address shouldn't block the extraction.

bodubas commented:
> I think according to other eyeExtend apps and [this dir generating tool](https://github.com/fs-connect/connect-images-dir/tree/master) we need to add a set of icons (connect, failed, gray, waiting) for all the actions

The thing is that those action images should be customized for each action so we have to involve designer for that, while it doesn't give much benefits. So to avoid delays, I've just added action group icon, without the actions.

Example of custom actions:
<img width="1440" alt="Screenshot 2023-10-26 at 18 22 29" src="https://github.com/KennaSecurity/eyeExtend-Connect/assets/67974435/026d0a01-caef-4af2-a034-cfe359a29c32">

How it's done for Cisco VM:
<img width="1440" alt="Screenshot 2023-10-26 at 18 22 24" src="https://github.com/KennaSecurity/eyeExtend-Connect/assets/67974435/46eabbb4-49f1-4cc8-9283-09a39a4f9676">


But, we don't want to get app returned from review based on that, so I might create some action items with arrows.

sveta-trotsiuk commented on CiscoVM/CiscoVM-1.0.0/property.conf:
```diff
@@ -70,18 +98,20 @@
       "display": "Cisco VM",
       "description": "Cisco Vulnerability Management templates",
-      "full_description": "<html>Use Cisco Vulnerability Management policy templates to manage devices.</html>",
+      "full_description": "<html>Use Cisco Vulnerability Management policy templates to export endpoints.</html>",
```
Maybe?: "export device state"

bodubas committed:
Add action images

bodubas commented:
<img width="613" alt="Screenshot 2023-10-27 at 16 19 04" src="https://github.com/KennaSecurity/eyeExtend-Connect/assets/67974435/0645db3d-a565-4bdc-9b05-7343ca744753">
Added icons for actions, removed background from logo.

bodubas committed:
Add space for encrypted field

bodubas committed:
Update README.md
